### PR TITLE
Fix limit=0 

### PR DESCRIPTION
**Pull request by philip1986**
This PR should fix the issue https://github.com/sequelize/sequelize/issues/3914.
When `option.limit` was set to `0` it was considered as limit isn't  set at all. This PR changes those checks to check for the existence of the limit option.

### CODE REVIEW

File: lib/dialects/abstract/query-generator.js [hunk 1]
```diff
@@ -1754,10 +1754,13 @@ var QueryGenerator = {
...
-    if (options.offset && !options.limit) {
+    var limitExist = typeof options.limit !== "undefined" && options.limit !== null;
+    var offsetExit = typeof options.offset !== "undefined" && options.offset !== null;
+
+    if (offsetExit && !limitExist) {
       fragment += ' LIMIT ' + this.escape(options.offset) + ', ' + 10000000000000;
-    } else if (options.limit) {
-      if (options.offset) {
+    } else if (limitExist) {
+      if (offsetExit) {
```
Comment (mickhansen) [inline lib/dialects/abstract/query-generator.js]: You can make this all a bit cleaner by using `options.limit != null`, a oddity of javascript allows you to check both undefined and null with that.

**Comment by mickhansen** [comment]
Please remove the merge commit, do a proper rebase instead.
`limit: 0` would result in no rows being returned, right?

**Comment by philip1986** [comment]
Ok, I will do do. Yes thats it will return an empty result set.

File: lib/dialects/abstract/query-generator.js (hunk 1 above)
Comment (philip1986) [inline lib/dialects/abstract/query-generator.js]: Unfortunately jshint settings doesn't allow `!= null`

**Comment by philip1986** [comment]
Will close this PR and reopen it after rebase.

**Pull request closed without merging**